src/hyperv/vmbus/vmbus_chan.c

```diff
@@ -390,9 +390,9 @@ vmbus_chan_gpadl_connect(struct vmbus_channel *chan, bus_addr_t paddr,
     }
     assert(page_count == 0); //invalid page count %d", page_count
 
-    msg = vmbus_msghc_wait_result(dev, mh);
-    status = ((const struct vmbus_chanmsg_gpadl_connresp *)
-        msg->msg_data)->chm_status;
+    const struct vmbus_chanmsg_gpadl_connresp *connresp =
+        (const struct vmbus_chanmsg_gpadl_connresp *)vmbus_msghc_wait_result(dev, mh);
```

Comment:
Also here, the pointer returned by vmbus_msghc_wait_result() is a `struct vmbus_message *`, then its `msg_data` field should be cast to a `struct vmbus_chanmsg_gpadl_connresp *`.